### ISSUE TITLE
Add more metrics

### ISSUE THREAD

**Issue by topepo** [tidymodels/yardstick#2]
- [ ] Kappa (weighted and ~unweighted~)
 - [ ] Somer's D
 - [ ] various log-losses, 
 - [ ] (c)rand statistic
 - [x] detection prevalence
 - [x] bal accuracy
 - [x] detection rate rate
 - [x] mape and similar regression metrics

**Comment by DavisVaughan** [comment]
A few more forecasting related ones:
- weighted mape
- symmetric mape
- mase (mean absolute scaled error)

https://en.m.wikipedia.org/wiki/Calculating_demand_forecast_accuracy

Even more contained in this post from Rob hyndman 

http://datascienceassn.org/sites/default/files/Another%20Look%20at%20Measures%20of%20Forecast%20Accuracy.pdf

**Comment by RobertKomara** [comment]
@topepo, You have already implemented MCC and Informedness (J-index). I am interested in coding up Markedness if you will take it?

**Comment by jonthegeek** [comment]
I need weighted kappa for something, so I'm going to work on that unless there's a reason not to do so (I prefer to be cautious when working on 2-year-old issues :) ).

**Comment by DavisVaughan** [comment]
So _generally_ we don't have an approach for adding sample weights to any metric yet.

That said, it looks like weighted kappa is a special case where there is just an option for weighting: `"linear"` or `"quadratic"`. Is this what you are looking for?

See the `weights` argument here, not `sample_weights`
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.cohen_kappa_score.html

And sort of how it works here:
https://stats.stackexchange.com/questions/59798/quadratic-weighted-kappa-versus-linear-weighted-kappa/206438

If that is what you want, a `kap_weighted(data, truth, estimate, weight = "linear", na_rm = TRUE, ...)` function would fit the current API and I could review a PR for that

**Comment by jonthegeek** [comment]
Yup, that's what I was thinking. TECHNICALLY you can use any weights you want in a kappa, but having those two options would work in every case I can think of.

It's definitely a weird case, but that's why it exists; it does a better job rating ordered factors than a normal kappa does.

Hopefully I can knock out the PR tomorrow. Thanks for the input!

**Comment by jonthegeek** [comment]
@DavisVaughan I started working on this, and now I'm wondering if it would hurt anything to add a "weight" parameter (default NULL) to `kap`, rather than splitting off a separate `kap_weighted` function (similar to what sklearn does). I realized that the help is exactly the same, just with an additional parameter to explain, and it doesn't look like it would hurt anything for that function to gain a parameter.

**Comment by jonthegeek** [comment]
Hmm, nevermind I think. For kap_weighted, it makes sense for the weight type to be before na_rm in the parameter list, which could break implementations of kap. So... keeping it separate.

**Comment by DavisVaughan** [comment]
I don't think there is much value in keeping this meta issue open anymore
